### PR TITLE
Credit Card Fraud detection end-to-end with XGBoost

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description
This PR contains end-to-end steps for XGBoost example
1) GenericTask, GenericController and ETLController, XGBoost DataLoader signature change (with optional 2nd arg) 
2) credit card data preparation scripts to expand data to 2 years
3) feature engineering step to enrich feature with 
4) pre-preprocess step to normalize the features  ( skip the categorical one-hot encoding) 
5) XGB training

For 2, 3, 4 steps, we have three separate notebooks to illustrate the steps, and three python files. the python files enable to run federated ETL process. 
3,4,5 steps are all constructed with new Job APIs. 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/advanced/finance-end-to-end/readme.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        "objective": "binary:logistic",
+        "eval_metric": "auc",
+        "tree_method": "hist",
+        "nthread": 16,
+    }
+
+    job = FedJob(name=job_name)
+
+    # Define the controller workflow and send to server
+    controller = XGBFedController()
```
Comment (YuanTingHsieh) [inline examples/advanced/finance-end-to-end/readme.md:292]: num_rounds=num_rounds,
xgb_params=xgb_params,
xgb_options={
  "early_stopping_rounds": early_stopping_rounds,
}

File: examples/advanced/finance-end-to-end/enrich_job.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+
+from nvflare import FedJob, ScriptExecutor
```
Comment (YuanTingHsieh) [inline examples/advanced/finance-end-to-end/enrich_job.py:18]: Not sure if this import ScriptExecutor statement still works after re-factoring of FedJob

File: nvflare/app_common/abstract/generic_task.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Dict, Optional
+
+from nvflare.app_common.abstract.fl_model import FLModel
+
+
+class GenericTask(FLModel):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/abstract/generic_task.py:21]: I prefer just call this `EmptyTask`? or `ControlTask` or `PingPongTask` or something that is more close to the usage of this.

File: nvflare/app_common/workflows/generic_task_controller.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from nvflare.app_common.abstract.generic_task import GenericTask
+from nvflare.app_common.workflows.model_controller import ModelController
+
+
+class GenericTaskController(ModelController):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/workflows/generic_task_controller.py:20]: or `PingPongController`?

File: examples/advanced/finance-end-to-end/enrich.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    # receives global message from NVFlare
+    etl_task = flare.receive()
+    merged_dfs = enrichment(input_dir, site_name)
+
+    for ds_name in merged_dfs:
+        save_to_csv(merged_dfs[ds_name], output_dir, site_name, ds_name)
+
+    # send message back the controller indicating end.
+    end_task = GenericTask()
+    flare.send(end_task)
```
Comment (YuanTingHsieh) [inline examples/advanced/finance-end-to-end/enrich.py:46]: should we just reuse the same task we receive so it is easier for user?
```
control_task = flare.receive()
xxx

flare.send(control_task)
```

**Review by YuanTingHsieh — COMMENTED** [review]
@ZiyueXu77 please check if the workflow is correct.

We also need to wait for the FedJob to be solid before merging this.

File: nvflare/app_common/abstract/generic_task.py (hunk 3 above)
Comment (ZiyueXu77) [reply nvflare/app_common/abstract/generic_task.py:21]: Instead of empty task, shall we have a "CommandTask":
params = {"command": "enrich"}
and at the end of the local run, it expect
FLModel(params={"task_status": "enrich_done"})?

File: examples/advanced/finance-end-to-end/feature_enrichment.ipynb [hunk 6]
```diff
@@ -0,0 +1,1359 @@
...
+     "output_type": "stream",
+     "text": [
+      "total 19596\n",
+      "drwxrwxr-x  2 chester chester    4096 Aug 23 16:08 \u001b[0m\u001b[01;34m.\u001b[0m/\n",
+      "drwxrwxr-x 12 chester chester    4096 Aug 23 16:07 \u001b[01;34m..\u001b[0m/\n",
+      "-rw-rw-r--  1 chester chester 5283276 Aug 23 16:07 history.csv\n",
+      "-rw-rw-r--  1 chester chester  961645 Aug 23 16:07 test.csv\n",
+      "-rw-rw-r--  1 chester chester 2240868 Aug 23 16:08 test_enrichment.csv\n",
+      "-rw-rw-r--  1 chester chester 3359157 Aug 23 16:07 train.csv\n",
+      "-rw-rw-r--  1 chester chester 8200671 Aug 23 16:08 train_enrichment.csv\n"
```
Comment (YuanTingHsieh) [inline examples/advanced/finance-end-to-end/feature_enrichment.ipynb:1273]: not sure if you want to show chester

**Review by YuanTingHsieh — COMMENTED** [review]
code-wise LGTM, @ZiyueXu77 please check the workflow thank you

**Comment by ZiyueXu77** [comment]
Looks good!

**Review by ZiyueXu77 — APPROVED** [review]
